packages/authorization-updater/src/readModelService.ts

```diff
@@ -97,27 +97,37 @@ export function readModelServiceBuilder(
       return undefined;
     },
 
-    async getAgreement(
+    async getLatestAgreement(
       eserviceId: EServiceId,
       consumerId: TenantId
     ): Promise<Agreement | undefined> {
-      const data = await agreements.findOne(
-        {
-          "data.eserviceId": eserviceId,
-          "data.consumerId": consumerId,
-          "data.state": {
-            $in: [
-              agreementState.active,
-              agreementState.archived,
-              agreementState.suspended,
-            ],
+      const data = await agreements
+        .find(
+          {
+            "data.eserviceId": eserviceId,
+            "data.consumerId": consumerId,
+            "data.state": {
+              $in: [
+                agreementState.active,
+                agreementState.archived,
+                agreementState.suspended,
+              ],
+            },
           },
-        },
-        { projection: { data: true } }
-      );
+          { projection: { data: true } }
+        )
+        .sort({ "data.createdAt": -1 })
+        .limit(1)
+        .toArray();
 
       if (data) {
-        const result = Agreement.safeParse(data.data);
+        if (data.length > 1) {
+          throw genericInternalError(
+            `Too many agreements returned: data ${JSON.stringify(data)} `
+          );
+        }
```

Comment:
Is this ever going to happen since we limit the result at only `1` (Either with `limit(1)` or `findOne`)?

Reply:
it should never happen, but since we do not have type safety and future changes may update one step but not the other, I kept the check just to be safe